source_code/main_mcu/src/fido2/crypto.c

```diff
@@ -0,0 +1,148 @@
+/*!  \file     crypto.c
+*    \brief    Crypto functions related to FIDO2
+*    Created:  15/10/2019
+*    Author:   0x0ptr@pm.me
+*/
+#include <stddef.h>
+#include <stdint.h>
+#include "rng.h"
+#include "platform_defines.h"
+#include "crypto.h"
+#include "bearssl_hash.h"
+#include "bearssl_hmac.h"
+#include "bearssl_ec.h"
+#include "bearssl_block.h"
+#include "bearssl_rand.h"
+#include "main.h"
+#include "sh1122.h"
+
+static br_sha256_context sha256_ctx;
```

Comment:
coding rules and global vars names.... though given this is static I'd say it's more about uniformity across all project files :)

Reply:
Agree

Reply:
actually, isn't "sha256_ctx" "snake case naming for functions and global variables"? Example of one of the existing function names: "debug_init_trace_buffer"...

Reply:
oh sorry i wasn't clear... I mean prepending var & function names with the name of the file they are in.

Reply:
oh, I see. The reason I was confused is that the coding rules does not specify this for vars. Only functions unless I missed it:

"function names should be prepended by the file name it is in"

But I will make the change. (and you should update the contribution guidelines :-)

Reply:
oops indeed! :)